otlp/common.go

```diff
@@ -83,10 +113,9 @@ func (ri *RequestInfo) ValidateTracesHeaders() error {
 	if ri.hasLegacyKey() && len(ri.Dataset) == 0 {
 		return ErrMissingDatasetHeader
 	}
-	switch ri.ContentType {
-	case "application/protobuf", "application/x-protobuf", "application/json":
+	if IsContentTypeSupported(ri.ContentType) {
 		return nil
-	default:
+	} else {
```

Comment:
nit: I prefer not to end a func with an else statement and maybe even have `return nil` the last statement to help it look cleaner.

eg I would end these functions with
```golang
if !IsContentTypeSupported(ri.ContentType) {
    return ErrInvaludContentType
}
return nil
```

Reply:
Is it too early to unify the headers validation logic? If not and we can do it now, I'd say keep the separate signal-specific named validation functions as the public API surface in case the logic diverges again in the future.

```go
// ValidateTracesHeaders validates required headers/metadata for a trace OTLP request
func (ri *RequestInfo) ValidateTracesHeaders() error {
	return validateHeaders(ri)
}

// ValidateMetricsHeaders validates required headers/metadata for a metric OTLP request
func (ri *RequestInfo) ValidateMetricsHeaders() error {
	return validateHeaders(ri)
}

// ValidateLogsHeaders validates required headers/metadata for a logs OTLP request
func (ri *RequestInfo) ValidateLogsHeaders() error {
	return validateHeaders(ri)
}

func validateHeaders(ri *RequestInfo) error {
	if len(ri.ApiKey) == 0 {
		return ErrMissingAPIKeyHeader
	}
	if ri.hasLegacyKey() && len(ri.Dataset) == 0 {
		return ErrMissingDatasetHeader
	}
	if !IsContentTypeSupported(ri.ContentType) {
		return ErrInvalidContentType
	}
	return nil // headers passed all the wickets!
}
```

Reply:
👍🏻 [Reorganized](https://github.com/honeycombio/husky/pull/121/commits/fdf0a4143a330411ec3fb90a5a8354f141a3a973) to a series of if-guards with the else case as the final statement for each validation function.

❓ Then I bonus'd on a [consolidation](https://github.com/honeycombio/husky/pull/121/commits/d59fcd3a578b80305f4c15f418b0449ffcfe29cb) to try it on. I like it, but it can be easily snipped from this PR if folks don't want to dry that up.

Reply:
I usually love the idea of consolidating instead of duplicating logic. However, I have concerns around shared validateHeader logic.

Metrics require dataset for both classic and current keys, at least for the foreseeable future (whereas traces require dataset for classic, service name for current).

Logs may have some [differing opinions](https://github.com/honeycombio/husky/issues/118) on whether to require service name or dataset (when not associated with a trace). I'm afraid we'd consolidate now just to split them back up again later.

Reply:
Fair enough! I'll open another PR undoing that consolidation.